google/cloud/bigtable/internal/completion_queue_impl.h

```diff
@@ -129,10 +129,15 @@ class AsyncUnaryRpcFuture : public AsyncGrpcOperation {
     return true;
   }
 
-  // While only one thread changes (or reads) the state of this class at a time,
-  // we need to ensure that the state changes are visible to other threads. An
-  // atomic is (in principle) a lighter weight mechanism to satisfy the memory
-  // order constraints.
+  // Only Start() and Notify() change and read the state of this class. The
+  // way we use gRPC guarantees that the thread calling Notify() does not call
```

Comment:
Again forgive my ignorance, but this sounds strange too.  If there is a guarantee that Notify() happens after Start(), then methinks there must be synchronization.